File: 0367. Valid Perfect Square/0367.myans.rs
impl Solution {
    pub fn is_perfect_square(num: i32) -> bool {
        let num = num as i64;
        let (mut left, mut right) = (1, num);

        while left <= right {
            let mid = left + (right - left) / 2;
            let square = mid * mid;
            if num == square {
                return true;
            } else if num > square {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        false
    }
}

File: 0367. Valid Perfect Square/0367.myans.go
func isPerfectSquare(num int) bool {
    left := 1
    right := num

    for left <= right {
        mid := left + (right - left) / 2
        square := mid * mid
        if num == square {
            return true
        } else if num < square {
            right = mid - 1
        } else {
            left = mid + 1
        }
    }

    return false
}